lstchain/high_level/tests/test_interpolate.py

```diff
@@ -0,0 +1,359 @@
+    )
+    irf_file_4 = simulated_dl2_file.parent / "en_dep_cut_irf_2.fits.gz"
+    run_program(
+        "lstchain_create_irf_files",
+        "--input-gamma-dl2",
+        simulated_dl2_file,
+        "--output-irf-file",
+        irf_file_4,
+        "--energy-dependent-gh",
+        "--energy-dependent-theta",
+        "--point-like",
+        "--gh-efficiency=0.8",
+        "--theta-containment=0.8",
+    )
+
+    irfs_1 = [simulated_irf_file, irf_file_2]
+    irfs_2 = [simulated_irf_file, simulated_irf_file]
+    irfs_3 = [irf_file_3, irf_file_4]
+    irfs_4 = [irf_file_3, irf_file_3]
+    irfs_5 = [simulated_srcdep_irf_file, simulated_srcdep_irf_file]
+
+    assert compare_irfs(irfs_1) == 0
+    assert compare_irfs(irfs_2)
+    assert compare_irfs(irfs_3) == 0
+    assert compare_irfs(irfs_4)
+    assert compare_irfs(irfs_5)
+
+
+def test_load_irf_grid(simulated_irf_file):
+    from lstchain.high_level.interpolate import load_irf_grid
+
+    aeff_list = load_irf_grid([simulated_irf_file], "EFFECTIVE AREA", "EFFAREA")
+
+    assert aeff_list.shape == (1, 23, 8)
+
+
+def test_interp_irf(simulated_irf_file, simulated_dl2_file):
+    from lstchain.high_level.interpolate import interpolate_irf
+
+    # Create another IRFs with different zenith and azimuth parameter
+    # Global cuts IRF
+    irf_file_g_2 = simulated_irf_file.parent / "irf_interp_0.fits.gz"
+    irf_file_g_3 = simulated_irf_file.parent / "irf_interp_1.fits.gz"
+    irf_file_g_final = simulated_irf_file.parent / "irf_interp_final.fits.gz"
+
+    hdus_2_g = [
+        fits.PrimaryHDU(),
+    ]
+    hdus_3_g = [
+        fits.PrimaryHDU(),
+    ]
+
+    # En-dep, point-like cuts IRF
+    irf_file_en = simulated_dl2_file.parent / "en_dep_cut_irf_1.fits.gz"
+
+    irf_file_en_2 = simulated_dl2_file.parent / "en_dep_cut_irf_2.fits.gz"
+    irf_file_en_3 = simulated_dl2_file.parent / "en_dep_cut_irf_3.fits.gz"
+    irf_file_en_final = simulated_dl2_file.parent / "interp_en_dep_cut_irf.fits.gz"
+
+    hdus_2_en = [
+        fits.PrimaryHDU(),
+    ]
+    hdus_3_en = [
+        fits.PrimaryHDU(),
+    ]
+
+    for irf in [simulated_irf_file, irf_file_en]:
+        # Change the effective area for different angular pointings
+        aeff_1 = Table.read(irf, hdu="EFFECTIVE AREA")
+        aeff_1_meta = fits.open(irf)["EFFECTIVE AREA"].header
+        aeff_2 = aeff_1.copy()
+        aeff_2_meta = aeff_1_meta.copy()
+
+        zen_1 = u.Quantity(aeff_1_meta["ZEN_PNT"], "deg").to_value(u.rad)
+        az_1 = u.Quantity(aeff_1_meta["AZ_PNT"], "deg").to_value(u.rad)
+        del_1 = u.Quantity(aeff_1_meta["B_DELTA"], "deg").to_value(u.rad)
+
+        zen_2 = 2 * zen_1
+        az_2 = 2 * az_1
+        del_2 = 1.2 * del_1
+
+        factor_zd = (np.cos(zen_2)) / np.cos(zen_1)
+        factor_del = (np.sin(del_2)) / np.sin(del_1)
+
+        aeff_1["EFFAREA"][0] *= factor_zd
+        aeff_2["EFFAREA"][0] *= factor_zd * factor_del
+
+        aeff_1_meta["ZEN_PNT"] = (zen_2 * 180 / np.pi, "deg")
```

Comment:
why zen2 here? Typo?

Reply:
Also in similar code blocks below

Reply:
The idea is that for `irf_1`, only the zenith is changed (doubled), and for `irf_2`, both zenith and delta values are changed. The `factor_*` is for the IRF values that 'should' have appropriate values assuming that the IRF varies in a linear way in this parameter space

Reply:
Ok. I will read exactly what is happening there later.